test/verify-test.ts

```diff
@@ -0,0 +1,240 @@
+import { ValidationError } from 'ripple-api/common/errors'
+import { verifyCommonFields } from './../src/models/transactions/common'
+import { assert } from 'chai'
+
+/**
+ * Transaction Verification Testing
+ *
+ * Providing runtime verification testing for each specific transaction type
+ */
+describe('Transaction Verification', function () {
+    it(`Verifies all optional CommonFields`, () => {
+        const txJson = {
+            Account: "r97KeayHuEsDwyU1yPBVtMLLoQr79QcRFe",
+            TransactionType: "Payment",
+            Fee: "12",
+            Sequence: 100,
+            AccountTxnID: "DEADBEEF",
+            Flags: 15,
+            LastLedgerSequence: 1383,
+            Memos: [
+                {
+                    Memo: {
+                        MemoType: "687474703a2f2f6578616d706c652e636f6d2f6d656d6f2f67656e65726963",
+                        MemoData: "72656e74"
+                    }
+                },
+                {
+                    Memo: {
+                        MemoFormat: "687474703a2f2f6578616d706c652e636f6d2f6d656d6f2f67656e65726963",
+                        MemoData: "72656e74"
+                    }
+                },
+                {
+                    Memo: {
+                        MemoType: "72656e74"
+                    }
+                }
+            ],
+            Signers: [
+                {
+                    Account: "r....",
+                    TxnSignature: "DEADBEEF",
+                    SigningPubKey: "hex-string"
+                }
+            ],
+            SourceTag: 31,
+            SigningPublicKey: "03680DD274EE55594F7244F489CD38CF3A5A1A4657122FB8143E185B2BA043DF36",
+            TicketSequence: 10,
+            TxnSignature: "3045022100C6708538AE5A697895937C758E99A595B57A16393F370F11B8D4C032E80B532002207776A8E85BB9FAF460A92113B9C60F170CD964196B1F084E0DAB65BAEC368B66"
+        }
+
+        assert.doesNotThrow(() => verifyCommonFields(txJson))
+    })
+
+    it(`Verifies only required CommonFields`, () => {
+        const txJson = {
+            Account: "r97KeayHuEsDwyU1yPBVtMLLoQr79QcRFe",
+            TransactionType: "Payment",
+        }
+        
+        assert.doesNotThrow(() => verifyCommonFields(txJson))
+    })
+
+    it (`Handles invalid Fee`, () => {
+        const invalidFee = {
+            Account: "r97KeayHuEsDwyU1yPBVtMLLoQr79QcRFe",
+            TransactionType: "Payment",
+            Fee: 1000
+        } as any
```

Comment:
Why is this cast `as any`?

Reply:
Because I wanted to test at runtime, and if its missing a field it won't be a type BaseTransaction